### PR TITLE
[SDKS-7067] Add new properties to BaseUpdate and genericMessageData

### PR DESCRIPTION
**Pull request by nmayorsplit**
# GO SPLIT COMMONS

## What did you accomplish?

- New feature: Instant feature flags

## Extra Notes

### CODE REVIEW

File: push/parser.go [hunk 1]
```diff
@@ -45,6 +46,29 @@ const (
...
 
+type compressType int64
+
+const (
+	NotCompressed compressType = 0
+	Gzip          compressType = 1
+	Zlib          compressType = 2
+)
+
+func (c compressType) String() *string {
```
Comment (mmelograno) [inline push/parser.go:57]: I believe that this is not needed

File: push/parser.go [hunk 2]
```diff
@@ -148,8 +172,11 @@ func (p *NotificationParserImpl) parseUpdate(data *genericData, nested *genericM
 	}
 
 	base := BaseUpdate{
-		BaseMessage:  BaseMessage{timestamp: data.Timestamp, channel: data.Channel},
-		changeNumber: nested.ChangeNumber,
+		BaseMessage:          BaseMessage{timestamp: data.Timestamp, channel: data.Channel},
+		changeNumber:         nested.ChangeNumber,
+		previousChangeNumber: nested.PreviousChangeNumber,
```
Comment (mmelograno) [inline push/parser.go:177]: `previousChangeNumber` `compressType` and the data should be instantiated only in `case UpdateTypeSplitChange:`

File: push/parser.go [hunk 3]
```diff
@@ -284,7 +311,10 @@ type Update interface {
 // BaseUpdate contains fields & methods related to update-based messages
 type BaseUpdate struct {
 	BaseMessage
-	changeNumber int64
+	changeNumber         int64
+	previousChangeNumber int64
```
Comment (mmelograno) [inline push/parser.go:315]: this is not BaseUpdate struct, it should be `type SplitChangeUpdate struct`

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: push/parser.go (hunk 1 above)
Comment (nmayorsplit) [reply push/parser.go]: I removed it.

File: push/parser.go (hunk 2 above)
Comment (nmayorsplit) [reply push/parser.go]: I moved these properties to SplitChangeUpdate.

File: push/parser.go (hunk 3 above)
Comment (nmayorsplit) [reply push/parser.go]: I moved them.

**Review by mmelograno — APPROVED** [review]

**Pull request merged**